Update sl-graphql.md

rpourzand opened the pull request:
Update with some of Will's graphql content

## What are you changing in this pull request and why?
<!---
Adding more documentation to graphql (not public yet but also released)

To learn more about the writing conventions used in the dbt Labs docs, see the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md).
-->

## Checklist
<!--
Uncomment if you're publishing docs for a prerelease version of dbt (delete if not applicable):
- [ ] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)
-->
- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `website/static/_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

rpourzand committed:
Update sl-graphql.md

Update with some of Will's graphql content

WilliamDee commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
````diff
@@ -105,3 +121,68 @@ order: [String!] = null
+
+```graphql
+compileSqlFromJinja(
+		environmentId: BigIn!
````
BigInt*

mirnawong1 replied:
adding @WilliamDee 's sugg

```suggestion
		environmentId: BigInt!
```

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -26,12 +26,20 @@ The dbt Semantic Layer GraphQL API allows you to explore and query metrics and d
 dbt Partners can use the Semantic Layer GraphQL API to build and integration with the dbt Semantic Layer.
 
+## Requirements to use the GraphQL API
+- A dbt cloud project on 1.6+
```
```suggestion
- A dbt cloud project on dbt v1.6 or higher
```

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -26,12 +26,20 @@ The dbt Semantic Layer GraphQL API allows you to explore and query metrics and d
+## Requirements to use the GraphQL API
+- A dbt cloud project on 1.6+
+- Metrics are defined and configured
+- A dbt Cloud Service Token that has semantic layer access
```
```suggestion
- A dbt Cloud [service token](/docs/dbt-cloud-apis/service-tokens) that has Semantic Layer access
```

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -26,12 +26,20 @@ The dbt Semantic Layer GraphQL API allows you to explore and query metrics and d
+- A dbt cloud project on 1.6+
+- Metrics are defined and configured
+- A dbt Cloud Service Token that has semantic layer access
+- Connection to the data warehouse is configured in the project
```
switch to active voice so it's easier for translation and nonenglish users

```suggestion
- Your dbt project is configured and connected to a data platform
```

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
````diff
@@ -88,9 +96,17 @@ metrics: [String!]!
 ```
 
-### Metric value query parameters
+### Metric Query
````
```suggestion
### Metric query
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -88,9 +96,17 @@ metrics: [String!]!
+
 
-The mutation is `createQuery`. The parameters are as follows:
+**Create Query**
```
```suggestion
**Create query**
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

mirnawong1 commented:
hey @rpourzand are we ready to merge this pr or not quite?

WilliamDee commented:
could we fix the indenting for the code blocks?

mirnawong1 committed:
Update sl-graphql.md

fix code indentions

mirnawong1 commented:
hey @WilliamDee great shout, have updated the indentation using the[this graphql formatter](https://goonlinetools.com/graphql-formatter/). let me know if it looks off!

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

rpourzand committed:
Update sl-graphql.md

- Removed start/end time
- Made a note about where filters no longer being a string (need to update)
- Made a note about adding examples when we're ready
- Made a note about the arrow --> Json format so we can add when ready 

This page is still in progress and shouldn't be published

rpourzand committed:
Update sl-graphql.md

Adding examples and json updates

rpourzand commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -88,20 +111,191 @@ metrics: [String!]!
-startTime: String = null
-endTime: String = null
-where: String = null
+where: [String] = null
```
@WilliamDee  is this correct now?

rpourzand replied:
Also, should we add Explain here?

WilliamDee replied:
Yup, the where should be list now. still abit of discussion on the explain, we may plan on adding compileSql endpoint instead of explain.

rpourzand replied:
Ah cool! Should we create a task for this in linear? I did see the explain one but not the compile one.

rpourzand commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -88,20 +111,191 @@ metrics: [String!]!
+    environmentId: <env_id>
+    metrics: ["food_order_amount", "order_gross_profit"]
+    groupBy: ["metric_time__month", "customer__customer_type)] - TODO update with syntax for object
+    limit: TODO
```
@WilliamDee  do you have an example of what limit/order looks like?

WilliamDee replied:
Both order and limit will work exactly the same as the JDBC one, so something like
- `limit: Optional[int]` ie., `limit: 10` for limiting the result to 10 rows
- `order: Optional[List[str]]` ie., `order=['-order_gross_profit', "metric_time__month"]` the `-` is for ordering in DESC

rpourzand replied:
@WilliamDee  thanks! wait so does order need the order name before the colon as well as the order outside of the list bracket?

I would assume it would be 

order: ["-revenue"] rather than order: order=['revenue']

rpourzand commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -88,20 +111,191 @@ metrics: [String!]!
+    environmentId: <env_id>
+    metrics: ["food_order_amount", "order_gross_profit"]
+    groupBy: ["metric_time__month", "customer__customer_type)] - TODO update with syntax for object
+    explain: TODO
```
@WilliamDee do you have an example with "explain?"

WilliamDee replied:
Reference the comment above about explain vs compileSql!

rpourzand committed:
Update sl-graphql.md

typos

rpourzand committed:
Update sl-graphql.md

fixing duplicates

rpourzand committed:
Update sl-graphql.md

adding limit and order examples

rpourzand committed:
Update sl-graphql.md

order/limit typo

rpourzand committed:
Update sl-graphql.md

adding data platform call

rpourzand committed:
Update sl-graphql.md

rpourzand committed:
Update sl-graphql.md

rpourzand committed:
Update sl-graphql.md

will and I went through and added some notes

rpourzand committed:
Update sl-graphql.md

A couple of more clean ups

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

rpourzand committed:
Update sl-graphql.md

Added a couple of notes on queryable grnaularities

WilliamDee committed:
updated GraphQL docs

rpourzand committed:
Update GQL docs (#3984)

Merge into @rpourzand branch for updating GQL with updated changes

WilliamDee commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -40,68 +48,301 @@ Authentication uses a dbt Cloud [service account tokens](/docs/dbt-cloud-apis/se
+    error
+    totalPages
+    arrowResult
+    jsonResult (orient: table)
```
this should be `orient: TABLE` uppercase

rpourzand committed:
Update sl-graphql.md

typo

rpourzand committed:
Update sl-graphql.md

fixing typo will caught

rpourzand commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
````diff
@@ -40,68 +48,301 @@ Authentication uses a dbt Cloud [service account tokens](/docs/dbt-cloud-apis/se
 ): [TimeGranularity!]!
 ```
 
+You can also get queryable granularities for all other dimensions using the `dimensions` call:
````
@courtneyholcomb  do you think you can add a "fetch what agg_time_dimension" example makes up metric time before or after this example? I wasn't quite sure how to structure it.

courtneyholcomb replied:
Yep I'll add a commit to this PR!

courtneyholcomb replied:
Done!

courtneyholcomb committed:
Add example for fetching measures & aggTimeDimension

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -40,68 +48,323 @@ Authentication uses a dbt Cloud [service account tokens](/docs/dbt-cloud-apis/se
+
+
+**Dimension Types**
+todo: add info on granularity when we expose it
```
hey @WilliamDee and @courtneyholcomb, this section seems like it's missing granularity info -  is that something either one of you can add please?

WilliamDee replied:
Oh woops, forgot to remove this, this is done right down below with the addition of
```
queryableGranularities: [TimeGranularity!]!
```

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
````diff
@@ -40,68 +48,323 @@ Authentication uses a dbt Cloud [service account tokens](/docs/dbt-cloud-apis/se
+
+```graphql
+{
+  query(environmentId: BigInt!, queryId: Int!, pageNum: Int {
````
hey @WilliamDee @courtneyholocomb - should pageNum: Int include an exclamation?

WilliamDee replied:
Nope it isn't required, it's defaulted to 1

Commit:
add graphql to other pages

Commit:
Merge branch 'rpourzand-graphql-updates' of https://github.com/dbt-labs/docs.getdbt.com into rpourzand-graphql-updates

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

mirnawong1 committed:
Update sl-graphql.md

remove 'todo' and fix json typo

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

mirnawong1 commented:
waiting until roxi is back from ooo

WilliamDee committed:
updated with new changes

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

rpourzand committed:
Update sl-graphql.md

adding the note about granularity output

rpourzand committed:
Update sl-graphql.md

adding a recommended sentence from @WilliamDee

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

rpourzand committed:
Update sl-graphql.md

@aiguofer Can you review this change I made in the docs to update the host for the explorer? I assume the schema explorer host will change once we do this.

WilliamDee committed:
update to add python example for querying

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -400,3 +400,63 @@ The results default to the table but you can change it to any [pandas](https://p
 
 By default, we return 1024 rows per page. If your result set exceeds this, you need to increase the page number using the `pageNum` option.
+
+### Execute a Query through Python
```
```suggestion
### Run a Python query
```

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -400,3 +400,63 @@ The results default to the table but you can change it to any [pandas](https://p
+
+### Execute a Query through Python
+
+The `arrowResult` from the GraphQL query response isn't all that useful visually as it's a byte dump. You can utilize any language that supports Arrow to convert that byte into a Arrow table. Here is an example of querying and decoding the arrow result in Python.
```
```suggestion
The `arrowResult` in the GraphQL query response is a byte dump, which isn't visually useful. You can convert this byte data into an Arrow table using any Arrow-supported language. Refer to the following Python example explaining how to query and decode the arrow result:

The `arrowResult` from the GraphQL query response isn't all that useful visually as it's a byte dump. You can utilize any language that supports Arrow to convert that byte into a Arrow table. Here is an example of querying and decoding the arrow result in Python.
```

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -400,3 +400,63 @@ The results default to the table but you can change it to any [pandas](https://p
+
+arrow_table = to_arrow_table(gql_response.json()["data"]["query"]["arrowResult"])
+
+# Perform whatever functionality is available e.g.,
```
```suggestion
# Perform whatever functionality is available, like convert to a pandas table.
```

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -400,3 +400,63 @@ The results default to the table but you can change it to any [pandas](https://p
+arrow_table = to_arrow_table(gql_response.json()["data"]["query"]["arrowResult"])
+
+# Perform whatever functionality is available e.g.,
+# Convert to a pandas table
```
```suggestion
```

mirnawong1 reviewed (approved):
hey @WilliamDee , made some small suggestions!

rpourzand commented:
Hey @mirnawong1 fyi - we are releasing the gql API on Monday!

mirnawong1 commented:
woohoo thanks for the heads up @rpourzand ! is this any time Monday or a specific one? if the latter, can you give me a ping so i can get this merged for ya'll 🙏

mirnawong1 commented:
going to resolve these conflicts (where did they come from!

mirnawong1 commented:
@WilliamDee , hope the suggests are ok with you. if so, I'll commit them monday

WilliamDee commented:
@mirnawong1 these all work! I could make the changes as well today since ill have abit of time!

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -40,68 +48,416 @@ Authentication uses a dbt Cloud [service account tokens](/docs/dbt-cloud-apis/se
+
+The `arrowResult` in the GraphQL query response is a byte dump, which isn't visually useful. You can convert this byte data into an Arrow table using any Arrow-supported language. Refer to the following Python example explaining how to query and decode the arrow result:
+
+The `arrowResult` from the GraphQL query response isn't all that useful visually as it's a byte dump. You can utilize any language that supports Arrow to convert that byte into a Arrow table. Here is an example of querying and decoding the arrow result in Python.
```
```suggestion
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 commented on website/docs/guides/dbt-ecosystem/sl-partner-integration-guide.md:
```diff
@@ -114,7 +114,7 @@ For better analysis, it's best to have the context of the metrics close to where
 
 **Stage 1 - The basic**
-* Supporting and using the new [JDBC](/docs/dbt-cloud-apis/sl-jdbc) is the first step. Refer to the [dbt Semantic Layer API](/docs/dbt-cloud-apis/sl-api-overview) for more technical details. 
+* Supporting and using the new [JDBC](/docs/dbt-cloud-apis/sl-jdbc) is the first step. Refer to the [dbt Semantic Layer APIs](/docs/dbt-cloud-apis/sl-api-overview) for more technical details. 
```
```suggestion
* Supporting and using the new [JDBC](/docs/dbt-cloud-apis/sl-jdbc) or [GraphQL](/docs/dbt-cloud-apis/sl-graphql) is the first step. Refer to the [dbt Semantic Layer APIs](/docs/dbt-cloud-apis/sl-api-overview) for more technical details. 
```

mirnawong1 committed:
Update website/docs/guides/dbt-ecosystem/sl-partner-integration-guide.md

mirnawong1 committed:
Merge branch 'current' into rpourzand-graphql-updates

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -22,16 +22,24 @@ GraphQL has several advantages, such as self-documenting, having a strong typing
 dbt Partners can use the Semantic Layer GraphQL API to build and integration with the dbt Semantic Layer.
 
+## Requirements to use the GraphQL API
+- A dbt cloud project on dbt v1.6 or higher
```
```suggestion
- A dbt Cloud project on dbt v1.6 or higher
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 commented on website/docs/docs/dbt-cloud-apis/sl-graphql.md:
```diff
@@ -22,16 +22,24 @@ GraphQL has several advantages, such as self-documenting, having a strong typing
+## Requirements to use the GraphQL API
+- A dbt Cloud project on dbt v1.6 or higher
+- Metrics are defined and configured
+- A dbt Cloud [service token](/docs/dbt-cloud-apis/service-tokens) that has Semantic Layer access
```
```suggestion
- A dbt Cloud [service token](/docs/dbt-cloud-apis/service-tokens with "Semantic Layer Only” and "Metadata Only" permissions
```

mirnawong1 committed:
Update website/docs/docs/dbt-cloud-apis/sl-graphql.md

mirnawong1 commented on website/docs/guides/dbt-ecosystem/sl-partner-integration-guide.md:
```diff
@@ -114,7 +114,7 @@ For better analysis, it's best to have the context of the metrics close to where
 
 **Stage 1 - The basic**
-* Supporting and using the new [JDBC](/docs/dbt-cloud-apis/sl-jdbc) is the first step. Refer to the [dbt Semantic Layer API](/docs/dbt-cloud-apis/sl-api-overview) for more technical details. 
+* Supporting and using the new [JDBC](/docs/dbt-cloud-apis/sl-jdbc) or [GraphQL](/docs/dbt-cloud-apis/sl-graphql) is the first step. Refer to the [dbt Semantic Layer APIs](/docs/dbt-cloud-apis/sl-api-overview) for more technical details. 
```
```suggestion
* Supporting and using [JDBC](/docs/dbt-cloud-apis/sl-jdbc) or [GraphQL](/docs/dbt-cloud-apis/sl-graphql) is the first step. Refer to the [dbt Semantic Layer APIs](/docs/dbt-cloud-apis/sl-api-overview) for more technical details. 
```

mirnawong1 committed:
Update website/docs/guides/dbt-ecosystem/sl-partner-integration-guide.md

mirnawong1 reviewed (approved):
graphql api is good to merge, merging now
